Sources/Testing/Parameterization/Test.Case.swift

```diff
@@ -95,15 +192,39 @@ extension Test {
       parameters: [Parameter],
       body: @escaping @Sendable () async throws -> Void
     ) {
-      let arguments = zip(values, parameters).map { value, parameter in
-        Argument(value: value, parameter: parameter)
+      // Attempt to obtain an encodable representation of each value in order
+      // to construct a stable ID.
+      let encodingResult = values.reduce(into: ([any Encodable](), hasFailure: false)) { result, value in
```

Comment:
Perhaps something like this might be clearer or easier to read?

```swift
let encodables: [any Encodable] = values.compactMap { value in
  ...
}
let hasFailure = encodables.count != values.count
```

Reply:
As I detailed in the comments, I want to bail early and stop attempting to get encodable versions of subsequent arguments if any earlier one failed. Getting an encodable value involves multiple conditional `as?` downcasts, and especially since you've mentioned some performance considerations here I wanted to be sure we're not doing unnecessary work. I know `reduce(into:)` can sometimes be less straightforward to read but I think my code comments explain everything adequately.

Reply:
Maybe we just do a plain old for-loop then?